packages/nft-erc1155-provider/lib/NftErc1155Provider.ts

```diff
@@ -0,0 +1,80 @@
+import { Provider } from '@liquality/provider'
+import { addressToString } from '@liquality/utils'
+import { ensure0x, normalizeTransactionObject } from '@liquality/ethereum-utils'
+import { NftProvider, Address } from '@liquality/types'
+
+import { Contract } from '@ethersproject/contracts'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { Wallet } from '@ethersproject/wallet'
+
+import NftErc1155_ABI from './NftErc1155_ABI.json'
+
+export default class NftErc1155Provider extends Provider implements Partial<NftProvider> {
+  _contract: Contract
+  _wallet: Wallet
+  _jsonRpcProvider: StaticJsonRpcProvider
+
+  constructor(options: { uri: string; mnemonic: string; derivationPath: string }) {
+    super()
+
+    this._wallet = Wallet.fromMnemonic(options.mnemonic, options.derivationPath)
+    this._wallet = this._wallet.connect(new StaticJsonRpcProvider(options.uri))
+    this._contract = new Contract('0x0000000000000000000000000000000000000000', NftErc1155_ABI, this._wallet)
+  }
+
+  private _attach(contract: Address | string) {
+    if (this._contract.address !== contract) {
+      this._contract = this._contract.attach(ensure0x(addressToString(contract)))
+    }
+  }
+
+  async balance(contract: Address | string, tokenID?: number) {
+    this._attach(contract)
+
+    if (tokenID) {
+      const amount = await this._contract.functions.balanceOf(await this._wallet.getAddress(), tokenID)
+      return amount[0].toNumber()
+    }
+
+    return 0
+  }
+
+  async transfer(
```

Comment:
I believe you have to check wether the `tokenIDs` is an array and if it's an array you should use `safeBatchTransferFrom` otherwise you should use `safeTransferFrom`

Reply:
Nice point.